ledger/internal/eval.go

```diff
@@ -94,15 +94,32 @@ type roundCowBase struct {
 	creators map[Creatable]ledgercore.FoundAddress
 }
 
-func makeRoundCowBase(l LedgerForCowBase, rnd basics.Round, txnCount uint64, compactCertNextRnd basics.Round, proto config.ConsensusParams) *roundCowBase {
-	return &roundCowBase{
+func makeRoundCowBase(l LedgerForCowBase, rnd basics.Round, txnCount uint64, compactCertNextRnd basics.Round, proto config.ConsensusParams, preloadedAccounts map[basics.Address]*basics.AccountData, preloadedCreators map[Creatable]ledgercore.FoundAddress) *roundCowBase {
+	if preloadedCreators == nil {
+		preloadedCreators = make(map[Creatable]ledgercore.FoundAddress)
+	}
+	rcb := &roundCowBase{
 		l:                  l,
 		rnd:                rnd,
 		txnCount:           txnCount,
 		compactCertNextRnd: compactCertNextRnd,
 		proto:              proto,
 		accounts:           make(map[basics.Address]basics.AccountData),
-		creators:           make(map[Creatable]ledgercore.FoundAddress),
+		creators:           preloadedCreators,
+	}
+	if preloadedAccounts != nil {
+		rcb.initializeAccountsCache(preloadedAccounts)
+	}
+	return rcb
+}
+
+func (x *roundCowBase) initializeAccountsCache(accts map[basics.Address]*basics.AccountData) {
+	for address, accountData := range accts {
+		if accountData == nil {
+			x.accounts[address] = basics.AccountData{}
```

Comment:
why can we have nil entries on init?

Reply:
The indexer would preload the touched accounts. If an account doesn't exist, it would place `nil` instead of an empty AccountData{}.

Reply:
can't we change this code in the indexer preloaded and get rid of `initializeAccountsCache` func from here?

Reply:
you mean implement the caching layer in `indexerLedgerConnector` instead ?